### PR TITLE
showing all the users who have the same rank as the top 10

### PR DESCRIPTION
**Pull request by dmbf29**
⚠️ This has big repercussions. In order for this to work, the [back-end PR](https://github.com/dmbf29/predictor-api/pull/121) needs to be merged as well.

Then in the front-end, we have access to all of the users. That way, if someone is ranked 12 but tied with the person who is 10th, we'll show them. And we also have access to the `currentUser`'s score and can display it if they're not in the top 10

### CODE REVIEW

**Comment by dmbf29** [comment]
<p float='left'>
image
image
</p>

**Comment by dmbf29** [comment]
⚠️ The biggest issue with this one is the results page. What do you want to have on it?

Without filtering it shows everyone in the app
image

With filtering, it shows no one, which is a bit strange
image

**Comment by dmbf29** [comment]
Maybe just show how the top 10 picked (plus the currentUser)?

**Comment by dmbf29** [comment]
So now it'll only show the top 10 (and whomever is logged in) on the results page

When you're not in the top 10 but made picks (currentUser is Julien)
image

When you're not in the top 10 but DIDNT made picks (currentUser is Aki)
image

When you're IN the top 10 (currentUser is Doug)
image

File: src/components/LeaderboardResults.vue [hunk 1]
```diff
@@ -53,18 +56,31 @@ export default {
           const results = {}
           Object.keys(this.leaderboard.results[match.id] ?? {}).forEach(
             choice => {
-              results[choice] = this.leaderboard.results[match.id][choice].map(
-                userId => {
-                  return this.leaderboard.users.find(u => u.userId === userId)
-                }
-              )
+              results[choice] = this.leaderboard.results[match.id][choice]
+                .map(userId => userId == this.currentUser.id ? this.rankedUsers.find(u => u.userId === userId) : this.topUsers.find(u => u.userId === userId))
```
Comment (Netsujr) [inline src/components/LeaderboardResults.vue:60]: ===

File: src/components/LeaderboardRankingsCard.vue [hunk 2]
```diff
@@ -50,20 +53,35 @@ export default {
...
+    sortedUsers: function () {
+      return this.leaderboard.users.slice().sort((a, b) => b.points - a.points)
+    },
+    isCurrentUserRanked() {
+      return this.rankedUsers.some(user => user.userId === this.currentUser.id && user.rank <= this.lastRank)
+    },
+    lastRank() {
+      if (this.rankedUsers.length === 0) return 0
+
+      if (typeof this.leaderboard.rankingsTopN === 'number') {
```
Comment (Netsujr) [inline src/components/LeaderboardRankingsCard.vue:65]: looks okay if we jst check `this.leaderboard.rankingsTopN`. If its null, hits the else

**Review by Netsujr — APPROVED** [review]

File: src/components/LeaderboardResults.vue (hunk 1 above)
Comment (dmbf29) [reply src/components/LeaderboardResults.vue]: 👍

File: src/components/LeaderboardRankingsCard.vue (hunk 2 above)
Comment (dmbf29) [reply src/components/LeaderboardRankingsCard.vue]: 👍

**Pull request merged**